airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -302,7 +305,8 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
     } else {
       final String failureReason = failureType == FailureType.CONFIG_ERROR ? "Connection Check Failed " + connectionId
           : "Job failed after too many retries for connection " + connectionId;
-      runMandatoryActivity(jobCreationAndStatusUpdateActivity::jobFailure, new JobFailureInput(connectionUpdaterInput.getJobId(), failureReason));
+      runMandatoryActivity(jobCreationAndStatusUpdateActivity::jobFailure,
```

Comment:
This will be an issue  because of temporal version. You need to use 
```
Workflow.getVersion(YOUR_OWN_TAG, Workflow.DEFAULT_VERSION, YOUR_CURRENT_VERSION);
```
If the version returned by this is >= YOUR_CURRENT_VERSION, you'll need to call 
```
runMandatoryActivity(jobCreationAndStatusUpdateActivity::jobFailure,
new JobFailureInput(connectionUpdaterInput.getJobId(), failureReason, attemptFailureSummary));
```
otherwise
`runMandatoryActivity(jobCreationAndStatusUpdateActivity::jobFailure, new JobFailureInput(connectionUpdaterInput.getJobId(), failureReason));`

This will only affect the current run and the right call will be made after when the job restarts.